### ISSUE TITLE
baryTest can produce nans in the singleton dimension

### ISSUE THREAD

**Issue by RussTedrake** [RobotLocomotion/drake#485]
Site: drake002
Build Name: master-RobotLocomotion-master
Build Server Directory: /tmp/drake-master-1299/drake
CDash: http://kobol.csail.mit.edu/cd/index.php?project=Drake&date=20141014

Most recent commits to drake included in this build:

```
fc15ebe Merge pull request #482 from RussTedrake/traj_opt_notes
5ca23e1 Merge pull request #474 from rdeits/quadrotor-3d-demo
2b5d4ab take geval back out of direct traj opt methods
d6c2152 Merge pull request #476 from kuindersma/rm-financial-toolbox-dep
1dda5f0 revert output argument order change in solveTraj
```

util/test/baryTest (unknown issue)

```
error_id = Drake:ValueCheck
error_message = NANs don't match. Found NANs at 
(3,1) (3,1) 
but desired_val has no NANs
is_segfault = 0
```

``` Matlab

                           < M A T L A B (R) >
                 Copyright 1984-2014 The MathWorks, Inc.
                   R2014a (8.3.0.532) 64-bit (glnxa64)
                            February 11, 2014


To get started, type one of these: helpwin, helpdesk, or demo.
For product information, visit www.mathworks.com.

To reproduce this test use rng(1504730857,'twister')

[Warning: Disabling autosave for simulink blocks (to avoid generating a lot
of
*.mdl.autosave files in your directory.  If you aren't a regular Simulink
user
and don't want this disabled, comment out this section in addpath_drake] 
[> In addpath_drake at 74
 In addpath_drake at 4] 

val =

    1   NaN     0
    0   NaN     0
    0   NaN     1

Error using valuecheck (line 78)
NANs don't match. Found NANs at 
(3,1) (3,1) 
but desired_val has no NANs

Error in baryTest (line 20)
 valuecheck(dpt,diag([1 0 1])); % zero because of the singleton dimension


<test_name>util/test/baryTest</test_name>
<error_id>Drake:ValueCheck</error_id> <error_message>NANs don't match.
Found NANs at 
(3,1) (3,1) 
but desired_val has no NANs</error_message> 


```

<matlab_test>strcmp(test_name,'util/test/baryTest') && ~isempty(strfind(error_message,'NANs'))</matlab_test>

**Comment by RussTedrake** [comment]
on quick inspection the random seeds displayed in the unit tests fail to reproduce the error on my machine or on the build server.  bummer.  will probably have to try a brute force (run it repeatedly until it fails locally) to capture it.

**Comment by jwnimmer-tri** [comment]
Closing as won't-fix, because #6369 will be removing the relevant software.

If I'm confused and this should in fact stay open, please re-open an clarify the remaining unsolved problem(s) in a new comment here.
